app/src/main/java/com/github/se/icebreakrr/ui/authentication/SignIn.kt

```diff
@@ -193,8 +205,19 @@ fun SignInScreen(
                 if (isTesting) {
                   navigationActions.navigateTo(TopLevelDestinations.AROUND_YOU)
                 } else {
-                  if (!NetworkUtils.isNetworkAvailable(context)) {
-                    NetworkUtils.showNoInternetToast(context)
+                  if (!NetworkUtils.isNetworkAvailable()) {
+                    if (hasAuthToken.value) {
+                      // If offline but has token, allow access
+                      navigationActions.navigateTo(TopLevelDestinations.AROUND_YOU)
+                      profilesViewModel.getFilteredProfilesInRadius(
```

Comment:
Question: Why calling this ? If it is in offline mode, nobody should be displayed in the aroundYou page.

Reply:
If I remember correctly, I did this because if I don't, when you signin without connectivity, the function that tries to reconnect every 5 seconds isn't immediately launched. But thanks to your comment I realized that my implementation is faulty because I set isConnected to false on launch in arroundYou if the user has no connectivity which stops the reconnection function from launching anyways.